framework/src/messaging/messaging_send.c

```diff
@@ -0,0 +1,192 @@
+		msgdbg("message send fail : sync recv fail %d.\n", errno);
+		MSG_FREE(sync_portname);
+		mq_close(sync_mqdes);
+		mq_unlink(sync_portname);
+		return ERROR;
+	}
+
+	MSG_FREE(sync_portname);
+	mq_close(sync_mqdes);
+	mq_unlink(sync_portname);
+
+	return OK;
+}
+
+static int messaging_send_internal(mqd_t *mqdes, const char *port_name, const char *msg, size_t msglen, int prio, void *data)
+{
+	int ret;
+	struct mq_attr internal_attr;
+
+	internal_attr.mq_maxmsg = CONFIG_MESSAGING_UNICAST_MAXMSG;
+	internal_attr.mq_msgsize = CONFIG_MESSAGING_UNICAST_MSGSIZE;
+	internal_attr.mq_flags = 0;
+
+	*mqdes = mq_open(port_name, O_WRONLY | O_CREAT, 0666, &internal_attr);
+	if (*mqdes == (mqd_t)ERROR) {
+		msgdbg("message send fail : open fail %d.\n", errno);
+		return ERROR;
+	}
+
+#ifdef CONFIG_MESSAGING_ENABLE_HEADER
+	ret = mq_send(*mqdes, (char *)data, sizeof(messaging_header_t) + msglen, prio);
+#else
+	ret = mq_send(*mqdes, msg, msglen, prio);
+#endif
+	if (ret != OK) {
+		msgdbg("message send fail : send fail %d.\n", errno);
+		return ERROR;
+	}
+	return OK;
+}
+
+/****************************************************************************
+ * messaging_send
+ ****************************************************************************/
+int messaging_send(const char *port_name, const char *msg, size_t msglen, msg_send_param_t *param)
+{
+	mqd_t mqdes;
+	int ret;
+
+	if (param->type == MSG_SEND_SYNC && param->sync_msg == NULL) {
+		msgdbg("message send fail : invalid param.\n");
+		return ERROR;
+	}
+
+#ifdef CONFIG_MESSAGING_ENABLE_HEADER
+	messaging_header_t data;
+	ret = messaging_append_header(msg, msglen, &data);
+	if (ret != OK) {
+		msgdbg("message send fail : out of memory.\n");
+		return ERROR;
+	}
+	ret = messaging_send_internal(&mqdes, port_name, msg, msglen, param->priority, &data);
+#else
+	ret = messaging_send_internal(&mqdes, port_name, msg, msglen, param->priority, NULL);
+#endif
+	if (ret != OK) {
+		goto send_fail;
+	}
+
+	if (param->type == MSG_SEND_SYNC) {
+		ret = messaging_sync_recv(port_name, param);
+		if (ret != OK) {
+			goto send_fail;
+		}
+	}
+
+	mq_close(mqdes);
+	return OK;
+
+send_fail:
+#ifdef CONFIG_MESSAGING_ENABLE_HEADER
+	MSG_FREE(data.msg);
+#endif
+	return ERROR;
+}
+
+int messaging_send_reply(const char *port_name, const char *msg, size_t msglen)
+{
+	int ret;
+	mqd_t mqdes;
+	char *reply_portname;
+
+	/* sender waits the reply with "port_name + MSG_SYNC_PORTNAME_PREFIX". */
+	MSG_ASPRINTF(&reply_portname, "%s%s", port_name, MSG_SYNC_PORTNAME_PREFIX);
+
+#ifdef CONFIG_MESSAGING_ENABLE_HEADER
+	messaging_header_t data;
+	ret = messaging_append_header(msg, msglen, &data);
+	if (ret != OK) {
+		msgdbg("message send fail : out of memory.\n");
+		return ERROR;
+	}
+	ret = messaging_send_internal(&mqdes, reply_portname, msg, msglen, MSG_REPLY_PRIO, &data);
+#else
+	ret = messaging_send_internal(&mqdes, reply_portname, msg, msglen, MSG_REPLY_PRIO, NULL);
+#endif
+	if (ret != OK) {
+		msgdbg("message reply fail.\n");
```

Comment:
I think you can change them like below
if (ret != OK) {
 ...
}
MSG_FREE
mq_close
return ret

Reply:
Done.